### ISSUE TITLE
Player object in space shooter bullet speeds are much slower and aren't registering hits.

### ISSUE THREAD

**Issue by GeminiSquishGames** [appsinacup/godot-rapier-2d#73]
Compared to the actions of default, when set to the Rapier plugin, bullets travel much slower. Regardless of speed they also seem to not trigger the usual methods on collisions as the default does. Setting the speed much higher seems to fix this but doesn't match the default use. For example, look in the repo I sent @Ughuuu an invite to, should be set to Default and has Rapier (slow version) installed. 

Edit: this and any other problems with speed can be fixed by:

` if ProjectSettings.get_setting("physics/2d/physics_engine") == "Rapier2D":`

and setting the values accordingly. Not sure what the magic number is but maybe something around [ speed *= 20 ] This may also affect other rigid bodies, like how fast the asteroids are when they spawn in, but it probably needs to match the speed of DEFAULT.

another edit: I will try to put a test scene in the res:// with easy bug reproductions so you can just test from there and not have to run the whole project to find the issues.

**Comment by Ughuuu** [comment]
Hm, the project is very big. Can you make just one scene with a single script if possible where this occurs?
You have multiple addons, some of which aren't signed(for mac) and I would rather not give access to them(eg libmgpt or how its called).
Also, after installing the rapier addon it just crashes for some reason. Anyway, a simpler project would make things easier to debug (eg just a rigidbody and apply_force or how you do it in script)

**Comment by Ughuuu** [comment]
Checking on the crash in the meanwhile by attaching debugger to see why it occurs.

**Comment by Ughuuu** [comment]
Ok, really weird it was a memory corruption error, if I delete the git plugin it works on my side and doesn't crash at least..

**Comment by Ughuuu** [comment]
Ok, so it seems the impulse apply slowness relates to the mass(the bullet has mass of 15, thats why it's 15x slower). The fix should be to make apply_impulse to take into account mass(on rapier wrapper side)

**Comment by GeminiSquishGames** [comment]
I'm working on making a smaller project right now. I wasn't sure what was easiest as I have to kinda pick apart what I've done since it seems to work with just simple things. The game is getting complex and there's probably more to it than just the general setup, also stuff that might be my fault so I'll see what I come up with. Will tackle bugs one at a time in the next few days. (Github addon has been strange and I probably just need to use a script to commit and do VC...)

**Comment by GeminiSquishGames** [comment]
I'll try to reproduce the bullet for it. I think I've found the problem with the Asteroids maybe, will put in another issue.

**Comment by Ughuuu** [comment]
On my tests it seems like it doesn't take into account mass. I know I made a change in the past for it to take into account mass, but then I changed it back for some reason. But it definitely is needed to keep into account the mass when calling apply_impulse. Will do a PR so you can test.

**Comment by Ughuuu** [comment]
Ok, after testing, it seems the bug is on godot side. I replicated the bug by doing:

```
var body = create_rigid_body
body.mass = 15
body.apply_central_impulse(Vector2(2000,0))
```

If I instead wait exatly 2 frames between setting mass 15 and applying central impulse, it works as intended. So the bug is on Godot Physics 2D server part, and in rapier it works slower(it's as intended actually).

**Comment by Ughuuu** [comment]
Create an issue on godot side here: https://github.com/godotengine/godot/issues/90899

**Comment by Ughuuu** [comment]
Well, I mean, godot also adds the mass to it, you can just multiply the force by the mass and use that. Godot multiplies but only after some time. So if 0.5s haven't passed, then godot will assume the mass is 1, even if you set it to 15kg.
Rapier takes into account the mass instantly, without delay. Thats why the object moves slower with rapier, because it correctly takes into account mass.
With godot, if you add a timer of 0.5s and then apply the impulse to the bullet, it will work same way(push it slower)

**Comment by GeminiSquishGames** [comment]
Heh, I mean to say add the equivalent force, not mass.

**Comment by Ughuuu** [comment]
Btw, you also said in the title: "and aren't registering hits". Is there also a demo related to that? Or was it just related to the force thing?

**Comment by GeminiSquishGames** [comment]
That seems to be happening with the Godot physics too now that I've tested
things more thoroughly. I'm going to assume it was related and refactor how
stuff is working after taking a look at it. I'll put up an issue if it
persists after that and some more testing. Thank you for your help.

**Comment by GeminiSquishGames** [comment]
Heh, the one thing about this is the little test project made me see how overcomplicated I was making things and relying on too many plugins... Starting over with Rapier at the get go seems to be working well.

**Comment by Ughuuu** [comment]
Also, idk why, but when testing your project I was reproducing a total system crash(on my macbook m2 pro) after clicking start game. Not sure why, also it was kind of heavy. I would say you could try to have no autoloads/globals and less things overall going on in menu/game. Also the sound effect seemed laggy, idk why. Thats what I saw.

**Comment by Ughuuu** [comment]
Depends also on the plugin, some plugins are really well, some are mostly buggy.

**Comment by GeminiSquishGames** [comment]
The crash, I'm not sure about at all but I don't have a way to test on mac os. Another reason I want to just kinda start over is the load, I noticed the way I was doing asteroids was kind of poor. The sound thing is probably Godot. I get the lag sometimes and have to fiddle with my focusrite's buffer size or unplug and plug back in. Hopefully these go away someday but I've always had issues with sound in Godot.
